Fix #4419: Remove TODO Marked Files And Other Alpha Kenya Related Files

XichengSpencer opened the pull request:
<!-- READ ME FIRST: Please fill in the explanation section below and check off every point from the Essential Checklist! -->
## Explanation
Fix #4419 
Removed application\alphakenya package as Adhiambo suggests.
Removed codes in bazel files,test_file_exemptions.textproto that is related with alphakenya 
Remove AlphaKenya related modules under android/util/logging including KenyaAlphaPlatformParameterModule, EventLoggingModule etc. and corresponding tests and Bazel configs
Inlined StandardEventTypeToHumanReadableNameConverterImpl.kt into EventTypeToHumanReadableNameConverter.kt
Removed EventLoggingConfigurationModule
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
<!-- Delete these section if this PR does not include UI-related changes. -->
If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- For PRs introducing new UI elements or color changes, both light and dark mode screenshots must be included
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

XichengSpencer committed:
Remove #4419 TODOs
Remove application\alphakenya package
Remove codes in bazel files,test_file_exemptions.textproto that is related with alphakenya
Remove AlphaKenya related modules and corresponding tests

XichengSpencer committed:
Remove the last file related with AlphaKenya

adhiamboperes commented:
@theMr17, could you please take an initial pass at this PR?

adhiamboperes commented:
@XichengSpencer, there are a number of merge conflicts in this PR, could you please try to resolve them?

theMr17 commented:
Hi @XichengSpencer, assign me the PR for review once the conflicts are resolved.

XichengSpencer committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into #4419

 Conflicts:
	app/src/main/java/org/oppia/android/app/application/alphakenya/AlphaKenyaApplicationComponent.kt
	app/src/main/java/org/oppia/android/app/application/alphakenya/BUILD.bazel
	domain/src/main/java/org/oppia/android/domain/platformparameter/PlatformParameterAlphaKenyaModule.kt
	utility/src/main/java/org/oppia/android/util/logging/BUILD.bazel
	utility/src/main/java/org/oppia/android/util/logging/KenyaAlphaEventTypeToHumanReadableNameConverterImpl.kt
	utility/src/test/java/org/oppia/android/util/logging/BUILD.bazel

adhiamboperes commented on .github/workflows/build_tests.yml:
```diff
@@ -422,7 +422,7 @@ jobs:
           bazel build --compilation_mode=opt -- //:oppia_alpha_kitkat
 
-      # Note that caching only works on non-forks.
+      # Note that caching only works on non-forks. 4419: should remove this?
```
Yes, should be removed, because this workflow will start failing now that the kenya build has been removed.

adhiamboperes commented on app/src/main/java/org/oppia/android/app/application/alphakenya/AlphaKenyaOppiaApplication.kt:
You need to also delete the BUILD and the AlphaKenyaApplicationComponent, and remove the package entirely.

adhiamboperes reviewed (changes requested):
Thanks @XichengSpencer! PTAL at the comment about files that still need to be removed, and also sync your branch with develop.

XichengSpencer committed:
Deleted Alpha Kenya package, PlatformParameterAlphaKenyaModule.kt, and workflow in build_tests.yml

XichengSpencer committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into #4419

 Conflicts:
	scripts/assets/test_file_exemptions.textproto
	version.bzl

XichengSpencer committed:
Delete more kenyaalpha related in BUILD.bazel,
Remove KenyaAlphaEventTypeToHumanReadableNameConverterImpl.kt
Resolved conflicts in test_file_exemptions.textproto

adhiamboperes reviewed (approved):
Thanks @XichengSpencer! This cleanup looks really good.

It seems you have some conflicts with develop, could you please resolve those by pulling develop into your local branch, then pushing after resolving any conflicts?

adhiamboperes commented:
It is also good practice to respond to all review comments that you have addressed, then assign the reviwer through the "assignees" section once done.

XichengSpencer committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into #4419

 Conflicts:
	utility/src/main/java/org/oppia/android/util/logging/KenyaAlphaEventTypeToHumanReadableNameConverterImpl.kt
	utility/src/test/java/org/oppia/android/util/logging/KenyaAlphaEventBundleCreatorTest.kt

XichengSpencer commented:
> It is also good practice to respond to all review comments that you have addressed, then assign the reviwer through the "assignees" section once done.

Thanks for letting me know Adhiambo!

XichengSpencer commented:
> Thanks @XichengSpencer! This cleanup looks really good.
> 
> It seems you have some conflicts with develop, could you please resolve those by pulling develop into your local branch, then pushing after resolving any conflicts?

Done!

XichengSpencer commented:
> It is also good practice to respond to all review comments that you have addressed, then assign the reviwer through the "assignees" section once done.

Sorry, I accidentally requested reviewer on the reviewer section, but I also did it on the assignees section.

adhiamboperes commented:
Thanks @XichengSpencer!

The PR LGTM. However, these changes require code owner approval, and both of the oppia/owners are out temporarily so we won't be able to merge this yet. If during merge there are conflicts to resolve or any other issue to address, @BenHenning will let you know.

Please feel free to work on other issues in the meantime.

Additionally, the "reviewer section" is automatically assigned, so you only ever need to change the "assignees section".

BenHenning commented on utility/src/main/java/org/oppia/android/util/logging/BUILD.bazel:
```diff
@@ -149,19 +149,6 @@ kt_android_library(
-    name = "kenya_alpha_event_logging_configuration_module",
-    srcs = [
-        "KenyaAlphaEventLoggingConfigurationModule.kt",
-        "KenyaAlphaEventTypeToHumanReadableNameConverterImpl.kt",
```
Can you also clean up this bit? We no longer need to have a separate interface/implementation for the event type to readable name converter. Specifically:

https://github.com/oppia/oppia-android/blob/develop/utility/src/main/java/org/oppia/android/util/logging/StandardEventTypeToHumanReadableNameConverterImpl.kt

Can be inlined into:

https://github.com/oppia/oppia-android/blob/develop/utility/src/main/java/org/oppia/android/util/logging/EventTypeToHumanReadableNameConverter.kt

I think this will allow removing EventLoggingConfigurationModule as well.

XichengSpencer replied:
I searched the usage for EventLoggingConfigurationModule which has 177 usages in tests and components. I just want to get an confirmation and I can remove them all. Hope removing them all won't cause any problem.

XichengSpencer replied:
Additional kenyaAlpha clean done, Inlined to [EventTypeToHumanReadableNameConverter.kt], and clean the EventLoggingConfigurationModule.

BenHenning replied:
Ah sorry--missed the earlier comment re: confirmation. I can confirm the approach looks correct!

BenHenning commented on version.bzl:
```diff
@@ -12,10 +12,8 @@ the app (that are potentially not broadly released yet).
 MINOR_VERSION = 13
 
-# TODO(#4419): Remove the Kenya-specific alpha version code.
 OPPIA_DEV_VERSION_CODE = 166
```
Let's go ahead and bump these versions right away so that there isn't a gap (that means setting ``OPPIA_GA_VERSION_CODE`` to 167 and then adding one for each version above it).

XichengSpencer replied:
Just double confirm you mean OPPIA_GA_VERSION_CODE to 161 or 167? Here are the current codes I have from latest develop branch
"OPPIA_DEV_VERSION_CODE = 166
OPPIA_DEV_KITKAT_VERSION_CODE = 165
OPPIA_ALPHA_VERSION_CODE = 163
OPPIA_ALPHA_KITKAT_VERSION_CODE = 162
OPPIA_BETA_VERSION_CODE = 161
OPPIA_GA_VERSION_CODE = 160"

XichengSpencer replied:
Version code updated

BenHenning replied:
The latest looks correct--thanks!

BenHenning reviewed (changes requested):
Thanks @XichengSpencer! It's great to see this get cleaned up. I had two additional suggestions (one of which is a consequential clean-up that we can do because of these changes). PTAL.

adhiamboperes commented:
@XichengSpencer, friendly ping to address the reviewer comments on this PR.

XichengSpencer committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into #4419

 Conflicts:
	utility/src/main/java/org/oppia/android/util/logging/KenyaAlphaEventTypeToHumanReadableNameConverterImpl.kt
	utility/src/test/java/org/oppia/android/util/logging/KenyaAlphaEventBundleCreatorTest.kt

XichengSpencer committed:
Remove EventLoggingConfigurationModule and updated the version code

XichengSpencer committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into #4419

 Conflicts:
	scripts/assets/test_file_exemptions.textproto

XichengSpencer committed:
Resolve conflict in test_file_exemptions.textproto

XichengSpencer committed:
Updated version.bzl and more clean up for EventLoggingConfigurationModule in test_file_exemptions.textproto & LogReportingModuleTest.kt

XichengSpencer committed:
Remove standard_event_logging_configuration_module in BUILD.bazel that caused build failed.

XichengSpencer committed:
More cleanup at EventTypeToHumanReadableNameConverterTest.

XichengSpencer committed:
Include javax in BUILD.bazel for EventTypeToHumanReadableNameConverter.kt
Added test BUILD.bazel config for EventTypeToHumanReadableNameConverterTest

XichengSpencer committed:
Fix syntax error in BUILD.bazel

XichengSpencer committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into #4419

adhiamboperes committed:
Merge branch 'develop' into #4419

adhiamboperes reviewed (changes requested):
Hi @XichengSpencer, could you please take a look at the following:
1. Static check failures - kdoc validity and link checks 
2. There seem to be some leftover modules in  some test classes, and the removed module appears to still be available in  a BUILD file.

Please make sure youb reassign the PR once done.

XichengSpencer committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into #4419

XichengSpencer committed:
Merge remote-tracking branch 'origin/#4419' into #4419

XichengSpencer committed:
Remove stard_event_logging_configuration part in BUILD.bazel and fix format issues

XichengSpencer commented:
> Hi @XichengSpencer, could you please take a look at the following:
> 
> 1. Static check failures - kdoc validity and link checks
> 2. There seem to be some leftover modules in  some test classes, and the removed module appears to still be available in  a BUILD file.
> 
> Please make sure youb reassign the PR once done.

Clean up done!

adhiamboperes committed:
Merge branch 'develop' into #4419

BenHenning commented on utility/src/main/java/org/oppia/android/util/logging/EventTypeToHumanReadableNameConverter.kt:
```diff
@@ -1,21 +1,88 @@
 
 /**
- * Converter for event types into loggable, human-readable names.
+ * Implementation of [EventTypeToHumanReadableNameConverter] which provides well-named
+ * representations for each event type.
```
```suggestion
 * Converter for event types into loggable, human-readable names.
```

Looks like this is out-of-date now that there's no longer an interface + multiple implementations. The summary fragment from the original interface seems suitable for this, now.

BenHenning commented on utility/src/test/java/org/oppia/android/util/logging/BUILD.bazel:
```diff
@@ -182,7 +109,22 @@ oppia_android_test(
+        "//third_party:org_robolectric_robolectric",
+        "//third_party:robolectric_android-all",
+    ],
+)
```
Nit: Please add back EOF newline.

BenHenning commented on scripts/assets/test_file_exemptions.textproto:
```diff
@@ -4362,10 +4342,6 @@ test_file_exemption {
-  override_min_coverage_percent_required: 0
-}
 test_file_exemption {
   exempted_file_path: "utility/src/main/java/org/oppia/android/util/logging/EventTypeToHumanReadableNameConverter.kt"
```
Can this exemption now be removed? You technically added a test for it by renaming the standard impl test.

XichengSpencer replied:
Weirdly, I could find this on my local. Because when I remove a file, I will do a global search to make sure all other appearance is removed. And it has happened more than once in this PR, just confused.
![image](https://github.com/user-attachments/assets/40ba1402-cecf-494e-a876-698289baebcc)

adhiamboperes replied:
@XichengSpencer, I think the exemption @BenHenning highlighted is for `EventTypeToHumanReadableNameConverter`

XichengSpencer replied:
> @XichengSpencer, I think the exemption @BenHenning highlighted is for `EventTypeToHumanReadableNameConverter`

Got it, thanks Adhiambo!

BenHenning reviewed (changes requested):
Thanks so much @XichengSpencer! This looks fantastic! Really sorry about the long review delay here.

I just had a few comments. Aside from those, can you please also update to the latest develop and fix the failing CI checks?

XichengSpencer committed:
Updated EventTypeToHumanReadableNameConverter.kt description

XichengSpencer committed:
Merge remote-tracking branch 'origin/#4419' into #4419

# Conflicts:
#	utility/src/main/java/org/oppia/android/util/logging/EventTypeToHumanReadableNameConverter.kt

XichengSpencer committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into #4419

XichengSpencer committed:
Fix Lint Error for BUILD.bazel

XichengSpencer committed:
Remove redundant test in BUILD.bazel
More cleanup for eventloggingconfigurationmodule

XichengSpencer committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into #4419

XichengSpencer committed:
Revert "Remove redundant test in BUILD.bazel More cleanup for eventloggingconfigurationmodule"

This reverts commit e0abb63f

XichengSpencer committed:
Remove redundant test in test_file_exemptions.textproto
More cleanup for eventloggingconfigurationmodule

XichengSpencer committed:
Fix Lint Error

XichengSpencer committed:
Merge branch '#4419' of https://github.com/XichengSpencer/oppia-android into #4419

XichengSpencer committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into #4419

XichengSpencer committed:
Lint error fixing

adhiamboperes reviewed (approved):
Thanks @XichengSpencer, I have looked at your latest changes and they seem to have addressed @BenHenning's comments. Handing over to @BenHenning to resplve.

BenHenning reviewed (approved):
Thanks @XichengSpencer! It's great to see this done. PR LGTM. :)

BenHenning commented:
Updating to latest develop and enabling auto-merge.

BenHenning committed:
Merge branch 'develop' into #4419
